### ISSUE TITLE
Failed connect to aaa.bbb.ccc.ddd:4443; Connection refused

### ISSUE THREAD

**Issue by itot555** [EMCECS/ECS-CommunityEdition#163]
I have executed following command to install ecs sw v300 on four node, that is CentOS 7.2.

`sudo python step1_ecs_multinode_install.py --ips 192.168.10.35 192.168.10.23 192.168.10.24 192.168.10.27 --hostnames ecsnode03 ecsnode01 ecsnode02 ecsnode04 --disks sdl sdm sdn sdo sdp sdq --ethadapter ens2f0`

However, failed to connect 192.168.10.35:4443 in spite of disabling firewalld service.
Below is output.

> [07/Oct/2016 20:09:28] INFO [root:628] Waiting on Authentication Service. This may take several minutes.
> Executing getAuthToken: curl -i -k https://192.168.10.35:4443/login -u root:ChangeMe
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed connect to 192.168.10.35:4443; Connection refused

Does anyone know how to fix that ?

**Comment by APeter0807** [comment]
Did you install on all four host VMs, or is this just the first? For a multi-node install, the authentication service and ECS portal won't become available until services on all four nodes are running concurrently.

There are other possible solutions outlined in Documentation/ECS-Troubleshooting.md

**Comment by itot555** [comment]
Thanks.

Not yet.
I have run step1_ecs_multinode_install.py script on only ecsnode03, so do I need to run this script on other 3 node ?

**Comment by APeter0807** [comment]
Yes; the installation won't be successful until the script is run on all four nodes.

**Comment by itot555** [comment]
I have already run step1 script on four node.

`[root@cent13 ecs-multi-node]# sudo python step1_ecs_multinode_install.py --ips 192.168.10.35 192.168.10.23 192.168.10.24 192.168.10.27 --hostnames ecsnode03 ecsnode01 ecsnode02 ecsnode04 --disks sdl sdm sdn sdo sdp sdq --ethadapter ens2f0`

`[root@cent14 ecs-multi-node]# sudo python step1_ecs_multinode_install.py --ips 192.168.10.35 192.168.10.23 192.168.10.24 192.168.10.27 --hostnames ecsnode03 ecsnode01 ecsnode02 ecsnode04 --disks sdl sdm sdn sdo sdp sdq --ethadapter ens2f0`

`[root@cent15 ecs-multi-node]# sudo python step1_ecs_multinode_install.py --ips 192.168.10.35 192.168.10.23 192.168.10.24 192.168.10.27 --hostnames ecsnode03 ecsnode01 ecsnode02 ecsnode04 --disks sdl sdm sdn sdo sdp sdq --ethadapter ens2f0`

`[root@cent17 ecs-multi-node]# sudo python step1_ecs_multinode_install.py --ips 192.168.10.35 192.168.10.23 192.168.10.24 192.168.10.27 --hostnames ecsnode03 ecsnode01 ecsnode02 ecsnode04 --disks sdl sdm sdn sdo sdp sdq --ethadapter ens2f0`

However, I think installation wasn't completed successfully and I can't access ECS portal yet.
 
```
[08/Oct/2016 08:48:54] INFO [root:645] Problem reaching authentication server. Retrying shortly.
[08/Oct/2016 08:48:54] CRITICAL [root:648] Authentication service not yet started.
[08/Oct/2016 08:48:54] INFO [root:777] Step 1 Completed.  Navigate to the administrator website that is available from any of the ECS data nodes.         The ECS administrative portal can be accessed from port 443. For example: https://ecs-node-external-ip-address.         The website may take a few minutes to become available.
[root@cent14 ecs-multi-node]# echo $?
0
```
Also, I haven't access below yet after four node installation.
Is there wrong installation operation ? 

```
[root@cent15 ecs-multi-node]# curl -i -k https://192.168.10.35:4443/login -u root:ChangeMe
curl: (7) Failed connect to 192.168.10.35:4443; Connection refused
```

**Comment by APeter0807** [comment]
You're getting "connection refused", which indicates that the service is up, but not able to use the correct port ( check `netstat -an | grep 443` ) . There are two likely causes:

Is firewalld running? If so, have you tried connecting with the firewall service disabled? If it's not running, have you tried with the firewall settings outlined in the documentation?

Are you running any service on these nodes such as an Apache server or previously-installed ECS container that might also be attempting to listen over ports 443 or 4443?

**Comment by itot555** [comment]
I haven't used firewall service on all four node.

```
[root@cent13 ecs-multi-node(swift)]# systemctl status firewalld
* firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; disabled; vendor preset: enabled)
   Active: inactive (dead)
```

Also, I have checked port usage on four all node, but 443 port isn't used in local server as following.

```
[root@cent13 ~(swift)]# netstat -an |grep 443
tcp6       0      0 127.0.0.1:64443         :::*                    LISTEN
unix  3      [ ]         STREAM     CONNECTED     10443    @/tmp/.ICE-unix/2716
```

```
[root@cent14 ~]# netstat -an |grep 443
tcp6       0      0 127.0.0.1:64443         :::*                    LISTEN
tcp6       0      0 :::443                  :::*                    LISTEN
tcp6       0      0 10.32.2.164:39443       10.32.210.114:9020      ESTABLISHED
```

```
[root@cent15 ~]# netstat -an |grep 443
tcp6       0      0 127.0.0.1:64443         :::*                    LISTEN
```

```
[root@cent17 ~(keystone)]# netstat -an |grep 443
tcp        0      0 10.32.2.167:47954       151.101.40.133:443      ESTABLISHED
tcp        0      0 10.32.2.167:48108       10.32.2.164:443         TIME_WAIT
tcp        0      0 10.32.2.167:48104       10.32.2.164:443         TIME_WAIT
tcp6       0      0 127.0.0.1:64443         :::*                    LISTEN
```

**Comment by APeter0807** [comment]
Try stopping a container and restarting it with `docker stop ecsmultinode` and `docker start -a ecsmultinode` - this may help indicate whether the issue is a duplicate of #147 , and even if not should show information on any components of the container that are failing during the startup process.

Additionally, any errors in the log for the authentication service ( `cat /var/log/vipr/emcvipr-object/authsvc.log | grep ERROR` )  may indicate issues particular to login.

**Comment by vijayscsa** [comment]
It's almost 5 days, still continuing with the problem of port 443 ( The management UI page is not at opening) after multiple clean installation and configuration.
with 2.2.1, and version 3 tried... but no luck till now.


[root@ilab1 ecs-single-node]# netstat -an | grep 443
tcp        0      0 0.0.0.0:64443           0.0.0.0:*               LISTEN
unix  3      [ ]         STREAM     CONNECTED     20443    /var/run/dbus/system_bus_socket
[root@ilabecsvm01 ecs-single-node]# netstat -lptn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:9099            0.0.0.0:*               LISTEN      18310/storageserver
tcp        0      0 0.0.0.0:9069            0.0.0.0:*               LISTEN      18310/storageserver
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1837/rpcbind
tcp        0      0 0.0.0.0:20048           0.0.0.0:*               LISTEN      1927/rpc.mountd
tcp        0      0 10.236.151.51:9204      0.0.0.0:*               LISTEN      17687/vnest
tcp        0      0 10.236.151.51:9205      0.0.0.0:*               LISTEN      17687/vnest
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1773/sshd
tcp        0      0 0.0.0.0:64443           0.0.0.0:*               LISTEN      18518/ecsportalsvc
tcp        0      0 0.0.0.0:42493           0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:55616           0.0.0.0:*               LISTEN      1806/rpc.statd
tcp        0      0 0.0.0.0:2049            0.0.0.0:*               LISTEN      -
tcp6       0      0 :::111                  :::*                    LISTEN      1837/rpcbind
tcp6       0      0 :::20048                :::*                    LISTEN      1927/rpc.mountd
tcp6       0      0 :::35318                :::*                    LISTEN      -
tcp6       0      0 :::22                   :::*                    LISTEN      1773/sshd
tcp6       0      0 :::2049                 :::*                    LISTEN      -
tcp6       0      0 :::50983                :::*                    LISTEN      1806/rpc.statd

[root@ilab1 ecs-single-node]# netstat -lptn | grep 443
tcp        0      0 0.0.0.0:64443           0.0.0.0:*               LISTEN      18518/ecsportalsvc


[root@ilab1 ecs-single-node]# cat /var/log/vipr/emcvipr-object/authsvc.log | grep ERROR
2016-11-07 08:53:48,860 [main] ERROR  DbClientImpl.java (line 210) Failed to start legacy coordinator for upgrade - this is safe to ignore in a non-upgrade scenario.
[root@ilabecsvm01 ecs-single-node]#

[root@ilab1 ecs-single-node]# docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
1bdb172135d2        33be14873330        "/opt/vipr/boot/boot."   8 months ago        Up 3 hours                              ecsstandalone


Need big help for this.

Thanks

**Comment by imYerik** [comment]
I met the same issuse. I resolved this by adding more RAM and CPU.
Check your ram and cpu status, one ecs node need 16G RAM and 4 cores at least.

**Comment by padthaitofuhot** [comment]
This is correct.  The minimum requirements for ECS reduced footprint build are 16GB RAM and 4 CPU cores.
